Tweak: Open modal when user wants to deactivate the Container experiment [ED-10091]

davseve opened the pull request.

davseve committed:
Tweak: Open modal when user wants to deactivate the Container experiment [ED-10091]

davseve committed:
WIP

davseve committed:
WIP

davseve committed:
Tweak: Open modal when user wants to deactivate the Container experiment [ED-10091]


WIP


WIP


WIP


WIP

davseve committed:
Merge remote-tracking branch 'origin/internal/ED-10091-open-modal-when-to-deactivating-exp' into internal/ED-10091-open-modal-when-to-deactivating-exp

davseve committed:
WIP

davseve committed:
WIP

davseve committed:
WIP

davseve commented on core/experiments/manager.php:
```diff
@@ -410,6 +410,15 @@ private function add_default_features() {
+			'messages' => [
+				// Translators: %1$s:Experiment title, %2$s: Experiment dependencies list
+				'on_deactivate' => sprintf( esc_html__(
+					'experiment will cuases all the existing containers to disappear /n
```
This text is just a place holder

davseve committed:
WIP

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -25,19 +25,26 @@ export default class ExperimentsDependency {
 		} );
 	}
 
+	inactiveStateActivity( experimentId ) {
```
need a better name

davseve replied:
Function removed

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -25,19 +25,26 @@ export default class ExperimentsDependency {
 	}
 
+	inactiveStateActivity( experimentId ) {
+		if ( !! this.isExperimentContainsDeactivatingMessage( experimentId ) && elementorAdminConfig.experiments[ experimentId ].state !== STATE_INACTIVE ) {
```
```suggestion
		if ( this.isExperimentContainsDeactivatingMessage( experimentId ) && elementorAdminConfig.experiments[ experimentId ].state !== STATE_INACTIVE ) {
```

davseve replied:
Done

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -25,19 +25,26 @@ export default class ExperimentsDependency {
 	}
 
+	inactiveStateActivity( experimentId ) {
+		if ( !! this.isExperimentContainsDeactivatingMessage( experimentId ) && elementorAdminConfig.experiments[ experimentId ].state !== STATE_INACTIVE ) {
```
extract the global variable access to local named variable (e.g. `initialState`, `originalState`, etc.)

davseve replied:
Function removed

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -25,19 +25,26 @@ export default class ExperimentsDependency {
 
 			case STATE_INACTIVE:
-				this.deactivateDependantExperiments( experimentId );
+				this.inactiveStateActivity( experimentId );
```
use the same convention as the previous case (if applicable)

davseve replied:
refactored to `shouldShowDeactivateModal`

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -58,6 +65,17 @@ export default class ExperimentsDependency {
 			) );
 	}
 
+	getExperimentDeactivatingDialogMessage( experimentId ) {
```
```suggestion
	getMessage( experimentId, messageId ) {
```

davseve replied:
What do you refer by `messageId`what should I pass?

davseve replied:
Done

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -58,6 +65,17 @@ export default class ExperimentsDependency {
+			.on_deactivate;
+	}
+
+	getExperimentMessages( experimentId ) {
+		return this.getExperimentData( experimentId ).messages;
+	}
```
remove

davseve replied:
Done

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,45 +127,29 @@ export default class ExperimentsDependency {
-	showDependenciesDialog( experimentId ) {
-		const experiment = this.getExperimentData( experimentId ),
-			dependencies = this.getExperimentDependencies( experimentId );
+	isExperimentContainsDeactivatingMessage( experimentId ) {
+		return this.getExperimentDeactivatingDialogMessage( experimentId );
+	}
```
```suggestion
	shouldShowDecativationDialog( experimentId ) {
		return !! this.getMessage( experimentId, 'on_deactivate' );
	}
```

davseve replied:
Done

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,45 +127,29 @@ export default class ExperimentsDependency {
+		// Translators: %1$s: Experiment title, %2$s: Message content
+		return __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
+			.replace( '%1$s', `<strong>${ title }</strong>` )
+			.replace( '%2$s', `<strong>${ message }</strong>` );
+	}
 
-		// Translators: %1$s: Experiment title, %2$s: Experiment dependencies list
-		const message = __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
-			.replace( '%1$s', `<strong>${ experiment.title }</strong>` )
-			.replace( '%2$s', `<strong>${ dependenciesList }</strong>` );
+	dialogDeactivateContentMessage( title, message ) {
+		// Translators: %1$s: Experiment title, %2$s: Message content
+		return __( 'While deactivating %1$s, %2$s.', 'elementor' )
+			.replace( '%1$s', `<strong>${ title }</strong>` )
+			.replace( '%2$s', `<strong>${ message }</strong>` );
+	}
```
maybe move to backend

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,45 +127,29 @@ export default class ExperimentsDependency {
+		// Translators: %1$s: Experiment title, %2$s: Message content
+		return __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
+			.replace( '%1$s', `<strong>${ title }</strong>` )
+			.replace( '%2$s', `<strong>${ message }</strong>` );
+	}
 
-		// Translators: %1$s: Experiment title, %2$s: Experiment dependencies list
-		const message = __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
-			.replace( '%1$s', `<strong>${ experiment.title }</strong>` )
-			.replace( '%2$s', `<strong>${ dependenciesList }</strong>` );
+	dialogDeactivateContentMessage( title, message ) {
+		// Translators: %1$s: Experiment title, %2$s: Message content
+		return __( 'While deactivating %1$s, %2$s.', 'elementor' )
+			.replace( '%1$s', `<strong>${ title }</strong>` )
+			.replace( '%2$s', `<strong>${ message }</strong>` );
+	}
```
maybe remove

davseve replied:
Refactored, added the functionality to `getMessageDialogConfig()` and `getDependencyDialogConfig()`

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -168,4 +170,60 @@ export default class ExperimentsDependency {
+	}
+
+	setDialogData( experiment, experimentId, state ) {
+		return state === STATE_INACTIVE
```
use regular if else, and then create build the messages inside it:

```JS
if ( active ) {
	// Translators: %1$s: Experiment title, %2$s: Message content
	const message = __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
			.replace( '%1$s', `<strong>${ title }</strong>` )
			.replace( '%2$s', `<strong>${ message }</strong>` );

	return {
		message,
		...
	}
} else {
	const experimentsList = this.joinDepenednciesNames( this.getExperimentDependencies( experimentId ).map( ( d ) => d.title ));

	const message = __( 'While deactivating %1$s, %2$s.', 'elementor' )
			.replace( '%1$s', `<strong>${ title }</strong>` )
			.replace( '%2$s', `<strong>${ message }</strong>` );
	return {
		 ...
	}
}
```

StyleShit replied:
maybe not relevant

davseve replied:
Not relevant

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,45 +127,29 @@ export default class ExperimentsDependency {
-				dependencies.forEach( ( dependency ) => {
-					this.setExperimentState( dependency.name, STATE_ACTIVE );
-				} );
+	showDialog( experimentId, state ) {
```
try to remove and use `this.getExperimentState()`

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -168,4 +170,60 @@ export default class ExperimentsDependency {
 		return clone.join( glue ) + finalGlue + lastItem;
 	}
+
+	setDialog( experiment, dialog ) {
+		return elementorCommon.dialogsManager.createWidget( 'confirm', {
+			id: 'e-experiments-messages-dialog',
+			headerMessage: dialog.headerMessage,
+			message: dialog.message,
+			position: {
+				my: 'center center',
+				at: 'center center',
+			},
+			strings: {
+				confirm: dialog.confirm,
+				cancel: dialog.cancel,
+			},
+			hide: {
+				onOutsideClick: false,
+				onBackgroundClick: false,
+				onEscKeyPress: false,
+			},
+			onConfirm: () => {
+				dialog.onConfirm();
+			},
+			onCancel: () => {
+				this.setExperimentState( experiment.name, dialog.initialState )
+			},
+		} );
+	}
```
maybe it's redundant, you can move it back to the original function

davseve replied:
DoneMoved in to `showDialog()` as suggested

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,45 +127,29 @@ export default class ExperimentsDependency {
-					this.setExperimentState( dependency.name, STATE_ACTIVE );
-				} );
+	showDialog( experimentId, state ) {
+		const experiment = this.getExperimentData( experimentId ),
+			dialogData = this.setDialogData( experiment, experimentId, state );
```
```suggestion
		const dialogData = isActive ? this.getActivationDialogData() : this.getDeactivationDialogData();
```

davseve replied:
Not relevant

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -25,19 +25,26 @@ export default class ExperimentsDependency {
 			case STATE_ACTIVE:
 				if ( this.shouldShowDependenciesDialog( experimentId ) ) {
-					this.showDependenciesDialog( experimentId );
+					this.showDialog( experimentId, STATE_ACTIVE );
```
```suggestion
					this.showDialog( this.getActivationDialogConfig( experimentId ) );
```

davseve replied:
Done

StyleShit commented on tests/jest/unit/core/experiments/assets/js/admin/behaviors/experiments-messages.test.js:
```diff
@@ -137,5 +137,69 @@ describe( 'ExperimentsDependency Behavior', () => {
+		deactivateExperiment( 'depends_on_active' );
+
+		// Assert.
+		expect( show ).not.toHaveBeenCalledTimes( 1 );
```
```suggestion
		expect( show ).not.toHaveBeenCalled()
```

StyleShit commented on core/admin/admin.php:
```diff
@@ -883,6 +883,7 @@ private function get_experiments() {
 					'state' => $experiment_data['state'],
 					'default' => $experiment_data['default'],
 					'dependencies' => $dependencies,
+					'messages' => $experiment_data['messages'] ?? [],
```
```suggestion
					'messages' => $experiment_data['messages'] ?? (object) [],
```
check if it works, and also change the mocks if it does

davseve replied:
Changed to object.

davseve committed:
Review comments fix

davseve committed:
WIP

Conditional changing added while checking experiment actual value due to the existence of hidden experiments

davseve committed:
WIP

davseve committed:
PHPUnit tests fixed

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -97,7 +115,7 @@ export default class ExperimentsDependency {
 				const isDependant = ( experimentData.dependencies.includes( experimentId ) );
 
-				if ( isDependant ) {
+				if ( isDependant && this.getExperimentActualState( id ) !== STATE_INACTIVE ) {
```
Added a variable.

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -168,4 +170,60 @@ export default class ExperimentsDependency {
+				initialState: STATE_ACTIVE,
+			}
+			: {
+				message: this.dialogActivateContentMessage( experiment.title, this.joinDepenednciesNames( this.getExperimentDependencies( experimentId ).map( ( d ) => d.title ) ) ) ,
```
Done

davseve committed:
Merge branch 'release/3.13.0' into internal/ED-10091-open-modal-when-to-deactivating-exp

davseve committed:
Review comments fix

davseve committed:
Merge remote-tracking branch 'origin/internal/ED-10091-open-modal-when-to-deactivating-exp' into internal/ED-10091-open-modal-when-to-deactivating-exp

davseve committed:
Merge branch 'release/3.13.0' into internal/ED-10091-open-modal-when-to-deactivating-exp

davseve committed:
playwright tests fixed

davseve committed:
Merge branch 'release/3.13.0' into internal/ED-10091-open-modal-when-to-deactivating-exp

StyleShit commented on core/experiments/manager.php:
```diff
@@ -410,6 +410,12 @@ private function add_default_features() {
 			'default' => self::STATE_INACTIVE,
+			'messages' => [
+				'on_deactivate' => esc_html__(
+					'you will permanently delete all content created with containers and lose access to container-based features like Tabs and Menu',
```
maybe just put the whole text here?
("if you deactivate ...")

davseve replied:
Done

StyleShit commented on tests/phpunit/elementor/core/admin/test-admin.php:
```diff
@@ -181,6 +181,8 @@ public function test_get_init_settings__returns_experiments_config() {
 				'title' => 'Experiment 1',
 				'state' => Experiments_Manager::STATE_ACTIVE,
 				'dependencies' => [],
+				'messages' => (object)[],
```
add a case with messages, to see that's being passed properly

davseve replied:
Turned to an array as we discussed earlier.

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -32,12 +32,16 @@ export default class ExperimentsDependency {
 			case STATE_INACTIVE:
-				this.deactivateDependantExperiments( experimentId );
+				if ( this.shouldShowDeactivateModal( experimentId ) ) {
+					this.showDialog( this.getMessageDialogConfig( experimentId ) );
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -58,6 +62,12 @@ export default class ExperimentsDependency {
 			) );
 	}
 
+	getMessage( experimentId, messageId = 'on_deactivate' ) {
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,43 +124,35 @@ export default class ExperimentsDependency {
-			dependencies = this.getExperimentDependencies( experimentId );
-
-		const dependenciesList = this.joinDepenednciesNames( dependencies.map( ( d ) => d.title ), ', ', ' & ' );
+	shouldShowDeactivateModal( experimentId ) {
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,43 +124,35 @@ export default class ExperimentsDependency {
-		const message = __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
-			.replace( '%1$s', `<strong>${ experiment.title }</strong>` )
-			.replace( '%2$s', `<strong>${ dependenciesList }</strong>` );
+		return !! this.getMessage( experimentId ) && initialState !== STATE_INACTIVE;
```
If the experiment's initial value is `inactive` and the user changes the select value to `default`, the deactivate dialog will pop up. This is not the desired behavior. 
We want to pop show the popup only when the experiment's initial value is inactive. 

Done :)

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,43 +124,35 @@ export default class ExperimentsDependency {
-				} );
-
-				this.elements.submit.click();
+				dialog.onConfirm();
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -168,4 +175,57 @@ export default class ExperimentsDependency {
+			experimentName = experiment.title,
+			dialogMessage = this.joinDepenednciesNames( this.getExperimentDependencies( experimentId ).map( ( d ) => d.title ) );
+
+		// Translators: %1$s: Experiment title, %2$s: Message content
```
Done

davseve commented on tests/playwright/pages/wp-admin-page.js:
```diff
@@ -107,6 +110,15 @@ module.exports = class WpAdminPage extends BasePage {
 		await this.page.locator( '#submit' ).click();
 	}
 
+	async ConfirmExperimentModalIfOpen() {
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -168,4 +175,57 @@ export default class ExperimentsDependency {
+			experimentName = experiment.title,
+			dialogMessage = this.getMessage( experimentId );
+
+		// Translators: %1$s: Experiment title, %2$s: Message content
+		const message = __( 'If you deactivate %1$s, %2$s.', 'elementor' )
```
The message is entirely passed from the backend.

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -168,4 +175,57 @@ export default class ExperimentsDependency {
+
+		return {
+			message,
+			headerMessage: __( 'Activate dialog.', 'elementor' ),
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,43 +124,35 @@ export default class ExperimentsDependency {
 			strings: {
-				confirm: __( 'Activate', 'elementor' ),
-				cancel: __( 'Cancel', 'elementor' ),
+				confirm: dialog.confirm,
+				cancel: dialog.cancel,
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -32,12 +32,16 @@ export default class ExperimentsDependency {
 			case STATE_ACTIVE:
 				if ( this.shouldShowDependenciesDialog( experimentId ) ) {
-					this.showDependenciesDialog( experimentId );
+					this.showDialog( this.getDependencyDialogConfig( experimentId ) );
```
Done.

davseve committed:
Review comments fix

davseve committed:
Merge branch 'release/3.13.0' into internal/ED-10091-open-modal-when-to-deactivating-exp

davseve committed:
JSLint fix

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,44 +118,33 @@ export default class ExperimentsDependency {
-		const message = __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
-			.replace( '%1$s', `<strong>${ experiment.title }</strong>` )
-			.replace( '%2$s', `<strong>${ dependenciesList }</strong>` );
+		return !! this.getMessage( experimentId, 'on_deactivate' ) && isActiveOfDefault;
```
```suggestion
		return hasMessage && isInitiallyActive;
```

davseve replied:
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -32,12 +32,16 @@ export default class ExperimentsDependency {
 			case STATE_ACTIVE:
 				if ( this.shouldShowDependenciesDialog( experimentId ) ) {
-					this.showDependenciesDialog( experimentId );
+					this.getDependencyDialogConfig( experimentId );
```
Done

davseve committed:
Review comments fix

davseve committed:
WIP

davseve committed:
Minor change

davseve committed:
WIP

StyleShit commented on tests/jest/unit/core/experiments/assets/js/admin/behaviors/experiments-messages.test.js:
```diff
@@ -202,5 +202,39 @@ describe( 'ExperimentsMessages Behavior', () => {
+
+		// Assert.
+		expect( getExperimentState( 'depends_on_inactive' ) ).toBe( 'inactive' );
+		expect( submitMock ).not.toHaveBeenCalled();
```
```JS
window.elementorCommon.dialogsManager.createWidget = jest.mock();
		
// ...

expect( window.elementorCommon.dialogsManager.createWidget ).not.toHaveBeenCalled();
```

StyleShit commented on tests/jest/unit/core/experiments/assets/js/admin/behaviors/experiments-messages.test.js:
```diff
@@ -202,5 +202,39 @@ describe( 'ExperimentsMessages Behavior', () => {
 		expect( submitMock ).not.toHaveBeenCalled();
 	} );
+
+	it( 'should not show modal when changing value from default to inactive while exp default value is inactive', () => {
```
```suggestion
	it( 'should not show modal when changing value from default to inactive while experiment default value is inactive', () => {
```
also others

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -120,9 +120,11 @@ export default class ExperimentsMessages {
 		const initialState = this.getExperimentData( experimentId ).state,
-			isActiveOfDefault = initialState !== STATE_INACTIVE;
+			defaultState = this.getExperimentData( experimentId ).default,
+			isInitiallyActive = initialState === STATE_ACTIVE || defaultState === STATE_ACTIVE,
```
```suggestion
			isInitiallyActive = initialState === STATE_ACTIVE || ( isDefault && defaultState === STATE_ACTIVE ),
```

davseve replied:
Done

StyleShit commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -120,9 +120,11 @@ export default class ExperimentsMessages {
-			isActiveOfDefault = initialState !== STATE_INACTIVE;
+			defaultState = this.getExperimentData( experimentId ).default,
+			isInitiallyActive = initialState === STATE_ACTIVE || defaultState === STATE_ACTIVE,
+		hasMessage = !! this.getMessage( experimentId, 'on_deactivate' );
```
```suggestion
			hasMessage = !! this.getMessage( experimentId, 'on_deactivate' );
```

davseve replied:
Done

davseve committed:
Test scenario added

davseve committed:
Merge branch 'release/3.13.0' into internal/ED-10091-open-modal-when-to-deactivating-exp

StyleShit reviewed (dismissed):
🔥 🔥

TzviRabinovitch commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,44 +118,35 @@ export default class ExperimentsDependency {
-		const experiment = this.getExperimentData( experimentId ),
-			dependencies = this.getExperimentDependencies( experimentId );
+	shouldShowDeactivationDialog( experimentId ) {
+		const initialState = this.getExperimentData( experimentId ).state,
```
maybe use experimentData= this.getExperimentData( experimentId );
and then access the state and default attributes instead of calling the function twice?

davseve replied:
Done :)

davseve committed:
Second Review comments fix

StuartElementor commented on tests/jest/unit/core/experiments/assets/js/admin/behaviors/experiments-messages.test.js:
```diff
@@ -137,5 +137,115 @@ describe( 'ExperimentsDependency Behavior', () => {
+
+	it( 'Should deactivate the experiment and its dependencies when clicking DEACTIVATE button', () => {
+		// Arrange.
+		const { confirm } = mockDialog();
+		const submitMock = jest.fn();
```
```suggestion
		const { confirm } = mockDialog(),
			submitMock = jest.fn();
```

StuartElementor commented on tests/jest/unit/core/experiments/assets/js/admin/behaviors/experiments-messages.test.js:
```diff
@@ -137,5 +137,115 @@ describe( 'ExperimentsDependency Behavior', () => {
+
+	it( 'Should not deactivate the experiment and its dependencies when clicking DISMISS button', () => {
+		// Arrange.
+		const { cancel } = mockDialog();
+		const submitMock = jest.fn();
```
```suggestion
		const { cancel } = mockDialog(),
			submitMock = jest.fn();
```

davseve replied:
Done

StuartElementor commented on tests/jest/unit/core/experiments/assets/js/admin/behaviors/experiments-messages.test.js:
```diff
@@ -137,5 +137,115 @@ describe( 'ExperimentsDependency Behavior', () => {
+
+	it( 'Should show the deactivate dialog when changing the select value from default to inactive while exp default value is active', () => {
+		// Arrange
+		const { confirm } = mockDialog();
+		const submitMock = jest.fn();
```
```suggestion
		const { confirm } = mockDialog(),
			submitMock = jest.fn();
```

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -109,44 +118,35 @@ export default class ExperimentsDependency {
+	shouldShowDeactivationDialog( experimentId ) {
+		const initialState = this.getExperimentData( experimentId ).state,
+			defaultState = this.getExperimentData( experimentId ).default,
+			isInitiallyActive = initialState === STATE_ACTIVE || ( initialState === STATE_DEFAULT && defaultState === STATE_ACTIVE ),
```
Done

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -95,9 +103,10 @@ export default class ExperimentsDependency {
 			.entries( elementorAdminConfig.experiments )
 			.forEach( ( [ id, experimentData ] ) => {
-				const isDependant = ( experimentData.dependencies.includes( experimentId ) );
+				const isDependant = ( experimentData.dependencies.includes( experimentId ) ),
```
Good one :)
Done

davseve committed:
Second Review comments fix (round 2)

davseve committed:
Merge branch 'release/3.13.0' into internal/ED-10091-open-modal-when-to-deactivating-exp

davseve commented on core/experiments/assets/js/admin/behaviors/experiments-messages.js:
```diff
@@ -168,4 +168,60 @@ export default class ExperimentsDependency {
+	showDependenciesDialog( experimentId ) {
+		const experiment = this.getExperimentData( experimentId ),
+			experimentName = experiment.title,
+			dialogMessage = this.joinDepenednciesNames( this.getExperimentDependencies( experimentId ).map( ( d ) => d.title ) );
```
Added an empty string as the `glue` default value.
